src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -207,11 +211,7 @@ def get_assignments(
             self.base_url + url,
         )
         while True:
-            current_page_assignments = assignments_per_section.current_page_items
-
-            for assignment in current_page_assignments:
-                assignment["section_id"] = section_id
-            assignments = assignments + current_page_assignments
+            assignments = assignments + assignments_per_section.current_page_items
```

Comment:
Though it is not part of this story, I refactored the assignment process in several places. Here in the `request_client` I decided not to assign the `section_id` - instead, only return what was received from the server. The `section_id` will be injected into the DataFrame via the mapper function.

Reply:
When I first developed this, the mapper did not exist, so not it makes sense to move it.